aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2525,7 +2666,8 @@ impl AptosVM {
                     Ok(None)
                 }
             },
-            TransactionPayload::Script(_)
+            TransactionPayload::BatchedCalls(_)
```

Comment:
How do batched calls work with randomness?

Reply:
So each call can invoke a public function in Move. For functions that uses randomness it has to be private entry function IIRC?